vignettes/readr.Rmd

```diff
@@ -160,60 +151,38 @@ readr comes with five parsers for rectangular file formats:
 * `read_fwf()` for fixed-width files
 * `read_log()` for web log files
 
-Each of these functions firsts calls `spec_xxx()` (as described above), and then parses the file according to that column specification:
+If readr finds problems with parsing your data, it will emit a warning message.
+For example, in our `challenge.csv` file, the column type we expected from our data was a column of dates and a column of doubles.
 
 ```{r}
-df1 <- read_csv(readr_example("challenge.csv"))
+df1 <- read_csv(readr_example("challenge.csv"), col_types = list(
+  date = col_date(),
+  value = col_double()
+))
```

Comment:
Note that this would still generate parsing errors even if we don't include/force `col_types` so we could just have the following if we think it makes for a better narrative:
```r
df1 <- read_csv(readr_example("challenge.csv"))
```
But I think it does drive the point across that supplying a `col_spec` is safer.